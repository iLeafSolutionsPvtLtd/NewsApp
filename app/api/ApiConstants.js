/* App config for apis
 */
const ApiConstants = {
  BASE_URL:
    "https://newsapi.org/v2/sources?apiKey=3883f1b24cb74b458d717c59a9cbb464",
  FIRST_URL: "https://newsapi.org/v2/everything?",
  LAST_URL:
    "from=2018-09-23&sortBy=publishedAt&apiKey=3883f1b24cb74b458d717c59a9cbb464"
};

export default ApiConstants;
