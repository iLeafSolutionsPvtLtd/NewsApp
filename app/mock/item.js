export default (item = {
  source: { id: "abc-news", name: "ABC News" },
  author: "The Associated Press",
  title: "Business Highlights",
  description:
    "Get breaking national and world news, broadcast video coverage, and exclusive interviews. Find the top news online at ABC news.",
  url: "https://abcnews.go.com/Business/wireStory/business-highlights-58294431",
  urlToImage: null,
  publishedAt: "2018-10-04T21:59:05Z",
  content:
    "Rising interest rates send stocks skidding; tech plunges NEW YORK (AP) Global stocks fell Thursday as interest rates in the U.S. continued to rise. Technology and internet companies skidded and the Nasdaq composite took its biggest loss in three months. Stronâ€¦ [+4919 chars]"
});
