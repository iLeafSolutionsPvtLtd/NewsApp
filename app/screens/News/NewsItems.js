import React, { Component } from "react";
import { View, Text, FlatList, Image, Dimensions } from "react-native";
import { Toolbar } from "react-native-material-ui";
import styles from "./styles";

export default class NewsItems extends Component {
  render() {
    const { item } = this.props;
    return (
      <View style={styles.itemContainer}>
        <Image
          style={styles.icon}
          source={{
            uri: item.urlToImage
          }}
        />
        <View style={styles.itemComponent}>
          <Text numberOfLines={3}>{item.publishedAt}</Text>
          <Text style={styles.title} numberOfLines={2}>
            {item.title}
          </Text>
          <Text numberOfLines={3}>{item.description}</Text>
        </View>
      </View>
    );
  }
}
