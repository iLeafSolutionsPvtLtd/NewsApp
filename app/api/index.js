// General api to access data
import ApiConstants from "./ApiConstants";
export function api(searchKey, source) {
  return fetch(
    `${ApiConstants.FIRST_URL}q=${searchKey}&sources=${source}&${
      ApiConstants.LAST_URL
    }`
  )
    .then(resp => resp.json())
    .then(json => json)
    .catch(error => error);
}
export function apiSource() {
  return fetch(ApiConstants.BASE_URL)
    .then(resp => resp.json())
    .then(json => json)
    .catch(error => error);
}
