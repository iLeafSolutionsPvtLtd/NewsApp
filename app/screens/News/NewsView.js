import React, { Component } from "react";
import {
  View,
  Text,
  FlatList,
  Image,
  Dimensions,
  TouchableOpacity
} from "react-native";

import ToolBar from "../../components/Toolbar";
import NewsItems from "./NewsItems";
import styles from "./styles";
var { height, width } = Dimensions.get("window");

class NewsView extends Component {
  _renderItem = ({ item }) => {
    return <NewsItems item={item} />;
  };

  render() {
    console.log("data here", this.props.data);
    return (
      <View style={styles.container}>
        <ToolBar {...this.props} />
        <View style={styles.component}>
          <FlatList
            data={this.props.data}
            extraData={this.props}
            numColumns={1}
            renderItem={this._renderItem}
          />
        </View>
      </View>
    );
  }
}

export default NewsView;
