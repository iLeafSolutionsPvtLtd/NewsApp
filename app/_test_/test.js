import "react-native";
import React from "react";
import NewsView from "../screens/News/NewsView";
import NewsItems from "../screens/News/NewsItems";
import ApiConstants from "../api/ApiConstants";
import { api } from "../api";
import mock_data from "../mock/news";
import item from "../mock/item";
import renderer from "react-test-renderer";

jest.mock("react-native-vector-icons/Feather", () => ({
  ToolbarAndroid: "ToolbarAndroid"
}));

it("renders correctly", () => {
  const tree = renderer.create(<NewsItems item={item} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Should load Routes", () => {
  global.fetch = jest.fn(
    () =>
      new Promise(resolve => {
        resolve({ status: 201, json: () => mock_data });
      })
  );

  return api("", "").then(data1 => {
    expect(data1).toBeDefined();
    expect(data1.status).toEqual("ok");
  });
});
