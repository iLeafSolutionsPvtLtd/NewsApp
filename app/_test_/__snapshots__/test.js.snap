// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "elevation": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": 333.5,
      "justifyContent": "center",
      "margin": 5,
      "padding": 10,
      "width": 730,
    }
  }
>
  <Image
    source={
      Object {
        "uri": null,
      }
    }
    style={
      Object {
        "flex": 1,
        "resizeMode": "cover",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 2,
        "marginLeft": 10,
      }
    }
  >
    <Text
      numberOfLines={3}
    >
      2018-10-04T21:59:05Z
    </Text>
    <Text
      numberOfLines={2}
      style={
        Object {
          "color": "black",
          "fontSize": 15,
          "fontWeight": "bold",
          "marginTop": 10,
        }
      }
    >
      Business Highlights
    </Text>
    <Text
      numberOfLines={3}
    >
      Get breaking national and world news, broadcast video coverage, and exclusive interviews. Find the top news online at ABC news.
    </Text>
  </View>
</View>
`;
